recipes/configs/llama3_1/kd_single_device.yaml

```diff
@@ -0,0 +1,110 @@
+# Config for single device knowledge distillation in kd_single_device.py
+# using a teacher and student model
+#
+# This config assumes that you've run the following commands before launching
+# this run:
+#   tune download meta-llama/Meta-Llama-3.1-8B-Instruct --output-dir /tmp/Meta-Llama-3.1-8B-Instruct
+#   tune run lora_finetune_single_device --config llama3_1/8B_lora_single_device
+#
+# To launch on a single device, run the following command from root:
+#   tune run kd_single_device --config llama3_1/kd_single_device
+#
+# This config works only for distilling on a single device.
+
+
+# Model Arguments
+model:
+  _component_: torchtune.models.llama3_1.lora_llama3_1
+  lora_attn_modules: ['q_proj', 'v_proj']
+  apply_lora_to_mlp: False
+  apply_lora_to_output: False
+  vocab_size: 128256
+  num_layers: 16
+  num_heads: 32
+  num_kv_heads: 8
+  embed_dim: 2048
+  max_seq_len: 131072
+  intermediate_dim: 8192
+  attn_dropout: 0.0
+  norm_eps: 1e-5
+  rope_base: 500000.0
+  lora_rank: 8
+  lora_alpha: 16
+  lora_dropout: 0.05
+  use_dora: False
+  quantize_base: False
+
+teacher_model:
+  _component_: torchtune.models.llama3_1.llama3_1_8b
+
+# Tokenizer
+tokenizer:
+  _component_: torchtune.models.llama3.llama3_tokenizer
+  path: /tmp/Meta-Llama-3.1-8B-Instruct/original/tokenizer.model
+  max_seq_len: null
+
+checkpointer:
+  _component_: torchtune.training.FullModelMetaCheckpointer
+  checkpoint_dir: /tmp/Llama-3.1-Student/
+  checkpoint_files: [
+    consolidated.00.pth
+  ]
+  recipe_checkpoint: null
+  output_dir: /tmp/kd_model/
+  model_type: LLAMA3
+
+# Teacher checkpoint
+teacher_checkpointer:
+  _component_: torchtune.training.FullModelMetaCheckpointer
+  checkpoint_dir: /tmp/Meta-Llama-3.1-8B-Instruct/lora_finetuned_single_device_epoch_1/
+  checkpoint_files: [
+    meta_model_0.pt
+  ]
+  recipe_checkpoint: null
+  output_dir: /tmp/Meta-Llama-3.1-8B-Instruct/
+  model_type: LLAMA3
+
+resume_from_checkpoint: False
+save_adapter_weights_only: False
+
+# Dataset and Sampler
+dataset:
+  _component_: torchtune.datasets.alpaca_cleaned_dataset
+seed: null
+shuffle: True
+batch_size: 8
+
+# Optimizer and Scheduler
+optimizer:
+  _component_: torch.optim.AdamW
+  weight_decay: 0.01
+  lr: 3e-4
+lr_scheduler:
+  _component_: torchtune.modules.get_cosine_schedule_with_warmup
+  num_warmup_steps: 100
+
+loss:
+  _component_: torchtune.modules.loss.CEWithChunkedOutputLoss
+
+kd_loss:
+  _component_: torchtune.modules.loss.ForwardKLWithChunkedOutputLoss
+kd_ratio: 0.5
+
+# Training
+epochs: 1
+max_steps_per_epoch: null
+gradient_accumulation_steps: 16
+compile: False
```

Comment:
Out of curiosity, did you try with compile yet?

Reply:
Yea tested that compile works (in pink)
<img width="364" alt="image" src="https://github.com/user-attachments/assets/f37487fe-68f3-47d5-a689-8c96c6b298a7">

Reply:
The loss curves between compile and non-compile are fairly similar.
<img width="349" alt="image" src="https://github.com/user-attachments/assets/c2012e80-d0e9-4f01-89f0-3e5b859b43f4"><img width="350" alt="image" src="https://github.com/user-attachments/assets/bd21e560-c0c6-4d40-a9cd-70476726657a">

I do see some non-determinism during eval. The losses are all around 1.2, but there's slight differences in eval. Also interesting that fine-tuning on alpaca dataset actually hurts performances on all benchmarks but Truthful QA.
<img width="793" alt="image" src="https://github.com/user-attachments/assets/4513b705-8a04-41c2-b873-b0ec83591022">